content/plugins/environment-plugin.md

```diff
@@ -1,24 +1,62 @@
 ---
 title: environment-plugin
 contributors:
-  - simon04
+  - einarlove
 ---
 
-The environment-plugin is a shorthand for using the [define-plugin](/plugins/define-plugin)
+The EnvironmentPlugin is a shorthand for using the [define-plugin](/plugins/define-plugin)
 on [`process.env`](https://nodejs.org/api/process.html#process_process_env) keys.
 
-**Example:**
+```js
+new webpack.EnvironmentPlugin(keys)
+```
+
+## How to use
+The EnvironmentPlugin accepts either an array of keys or an object with default values.
+
+```js
+new webpack.EnvironmentPlugin(['NODE_ENV', 'VERSION'])
+```
 
-```javascript
-new webpack.EnvironmentPlugin([
-  'NODE_ENV'
-])
+When given an object, the values are used as defaults for when they are not defined in `process.env`.
+
+```js
+new webpack.EnvironmentPlugin({
+  NODE_ENV: 'development',
+  VERSION: 33,
+})
 ```
 
-This is equivalent to the following define-plugin application:
+T> Warning is gived if the value is undefined. This can be suppressed with a default value or a value of `null`.
 
-```javascript
-new webpack.DefinePlugin({
-  'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV)
+## Example
```

Comment:
I believe the docs for EnvironmentPlugin should describe how it works and be understood without knowing how DefinePlugin works. There's no need to be sparse on examples.